pkg/backup/backup.go

```diff
@@ -489,31 +535,30 @@ func (kb *kubernetesBackupper) BackupWithResolvers(
 		addNextToBlock := i < len(items)-1 && items[i].orderedResource && items[i+1].orderedResource && items[i].groupResource == items[i+1].groupResource
 		if itemBlock != nil && len(itemBlock.Items) > 0 && !addNextToBlock {
 			log.Infof("Backing Up Item Block including %s %s/%s (%v items in block)", items[i].groupResource.String(), items[i].namespace, items[i].name, len(itemBlock.Items))
-			backedUpGRs := kb.backupItemBlock(ctx, *itemBlock)
-			for _, backedUpGR := range backedUpGRs {
-				backedUpGroupResources[backedUpGR] = true
+			wg.Add(1)
+			backupRequest.ItemBlockChannel <- ItemBlockInput{
+				itemBlock:  itemBlock,
+				returnChan: itemBlockReturn,
 			}
 			itemBlock = nil
 		}
+	}
 
-		// updated total is computed as "how many items we've backed up so far, plus
-		// how many items we know of that are remaining"
-		backedUpItems := backupRequest.BackedUpItems.Len()
-		totalItems := backedUpItems + (len(items) - (i + 1))
-
-		// send a progress update
-		update <- progressUpdate{
-			totalItems:    totalItems,
-			itemsBackedUp: backedUpItems,
-		}
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		wg.Wait()
```

Comment:
This thread will stay around even after a cancellation, I think, because there will be items in the wait group, but you will have stopped the processing thread.

Reply:
After further investigation (and discussion with @shawn-hurley ) we have determined that this is fine. The only scenario that would leave this thread hanging around would be if `cancelFunc` were called from outside of the current `BackupWithResolvers` function, since the only place we're calling it here is after we've already waited for the `done` channel to close. However, `cancelFunc` is only ever assigned to a local variable, so it's not possible for something outside to call it. Therefore this is safe.

Reply:
> There is a minor nit here. I need to use a different var name for cancelFunc since this name is also used earlier in this scope for the podVolumeBackup timeout, so I need to use a different name here to avoid variable overwriting. I'll do that on update.

Yeah I was thinking about the same yesterday, whether the same name would be problematic or not 😓

Reply:
@shubham-pampattiwar it was a simple oversight. I changed the name of the context var but forgot to update the cancelFunc var name. Forthcoming PR update will fix it.